pkg/kubernetes/nodes.go

```diff
@@ -96,3 +108,266 @@ func (k *Kubernetes) NodesTop(ctx context.Context, options NodesTopOptions) (*me
+	SourcePath string
+	DestPath   string
+	Namespace  string
+	Image      string
+	Privileged bool
+}
+
+// NodesFiles handles file operations on a node filesystem by creating a privileged pod
+func (k *Kubernetes) NodesFiles(ctx context.Context, opts NodeFilesOptions) (string, error) {
+	// Set defaults
+	if opts.Namespace == "" {
+		opts.Namespace = "default"
+	}
+	if opts.Image == "" {
+		opts.Image = "busybox"
+	}
+
+	// Create privileged pod for accessing node filesystem
+	podName := fmt.Sprintf("node-files-%s", rand.String(5))
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{APIVersion: "v1", Kind: "Pod"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: opts.Namespace,
+			Labels: map[string]string{
+				AppKubernetesName:      podName,
+				AppKubernetesComponent: "node-files",
+				AppKubernetesManagedBy: version.BinaryName,
+			},
+		},
+		Spec: v1.PodSpec{
+			NodeName:      opts.NodeName,
+			RestartPolicy: v1.RestartPolicyNever,
+			Containers: []v1.Container{{
+				Name:    "node-files",
+				Image:   opts.Image,
+				Command: []string{"/bin/sh", "-c", "sleep 3600"},
+				SecurityContext: &v1.SecurityContext{
+					Privileged: ptr.To(opts.Privileged),
+				},
+				VolumeMounts: []v1.VolumeMount{{
+					Name:      "node-root",
+					MountPath: "/host",
+				}},
+			}},
+			Volumes: []v1.Volume{{
+				Name: "node-root",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: "/",
+					},
+				},
+			}},
+		},
+	}
+
+	// Create the pod
+	pods, err := k.AccessControlClientset().Pods(opts.Namespace)
+	if err != nil {
+		return "", fmt.Errorf("failed to get pods client: %w", err)
+	}
+
+	createdPod, err := pods.Create(ctx, pod, metav1.CreateOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to create pod: %w", err)
+	}
+
+	// Ensure pod is deleted after operation
+	defer func() {
+		deleteCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+		_ = pods.Delete(deleteCtx, podName, metav1.DeleteOptions{})
+	}()
+
+	// Wait for pod to be ready
+	if err := k.waitForPodReady(ctx, opts.Namespace, podName, 2*time.Minute); err != nil {
+		return "", fmt.Errorf("pod failed to become ready: %w", err)
+	}
+
+	// Perform the requested operation
+	var result string
+	var opErr error
+	switch opts.Operation {
+	case "put":
+		result, opErr = k.nodeFilesPut(ctx, opts.Namespace, podName, opts.SourcePath, opts.DestPath)
+	case "get":
+		result, opErr = k.nodeFilesGet(ctx, opts.Namespace, podName, opts.SourcePath, opts.DestPath)
+	case "list":
+		result, opErr = k.nodeFilesList(ctx, opts.Namespace, podName, opts.SourcePath)
+	default:
+		return "", fmt.Errorf("unknown operation: %s", opts.Operation)
+	}
+
+	_ = createdPod
+	return result, opErr
+}
+
+// nodeFilesPut copies a file from local filesystem to node filesystem
+func (k *Kubernetes) nodeFilesPut(ctx context.Context, namespace, podName, sourcePath, destPath string) (string, error) {
```

Comment:
For the copy operations we should probably replicate what kubectl does to ensure the behavior is consistent:

https://github.com/kubernetes/kubectl/blob/master/pkg/cmd/cp/cp.go

Personally, I've had some bad experiences with the implementation of these operations in the [Fabric8 Kubernetes Client](https://github.com/fabric8io/kubernetes-client/)